app/controllers/users_controller.rb

```diff
@@ -1,40 +1,90 @@
 # frozen_string_literal: true
 
-# TODO: move this into a new UserController
-# see observer_controller.rb
-module ObserverController::UserController
+class UsersController < ApplicationController
+  # These need to be moved into the files where they are actually used.
+  require "find"
+  require "set"
+
+  before_action :login_required
+  before_action :disable_link_prefetching, except: [
+    :show,
+    :by_contribution
+  ]
+
   # User index, restricted to admins.
-  def index_user
+  def index
+    return patterned_index if params[:pattern].present?
+
     if in_admin_mode? || find_query(:User)
       query = find_or_create_query(:User, by: params[:by])
       show_selected_users(query, id: params[:id].to_s, always_index: true)
     else
       flash_error(:runtime_search_has_expired.t)
-      redirect_to(action: "list_rss_logs")
+      redirect_to("/")
     end
   end
 
+  alias index_user index
   # People guess this page name frequently for whatever reason, and
   # since there is a view with this name, it crashes each time.
-  alias list_users index_user
+  alias list_users index
 
   # User index, restricted to admins.
-  def users_by_name
+  def by_name
     if in_admin_mode?
       query = create_query(:User, :all, by: :name)
       show_selected_users(query)
     else
       flash_error(:permission_denied.t)
-      redirect_to(action: "list_rss_logs")
+      redirect_to("/")
     end
   end
 
+  # by_contribution.rhtml
+  def by_contribution
```

Comment:
Oh yes, if it's using SiteData, it's got to be expensive, therefore we should not allow it to be prefetched.